EyeSeeTea/dhis2-light-sdk
# Feature add program endpoint

### :pushpin: References
* **Issue:** close https://github.com/EyeSeeTea/malariapp/issues/2241
* **Related pull-requests:** https://github.com/EyeSeeTea/malariapp/pull/2242

### :tophat: What is the goal?

Create request to endoint /programs 

###   :gear: branches 
**dhis2-light-sdk**:
       Origin: add_organisation_program Target: developent

### :memo: How is it being implemented?

* I have added program endpoint class, I have created models and tests for clients retrieving programs 
* Second I have update androidSample and webSample

### :boom: How can it be tested?

 **Use case 1:** execute code analysis and execute all tests with this command:
```
./gradlew build
```

**Use case 2:** execute Android sample app, insert url, ApiVersion, username, password, select program endpoint, insert ids to retrieve and click on the button to realize request against dhis2 server. The response toString method should be rendered on the screen

**Use case 5:** execute web sample app, insert url, ApiVersion, username, password, select program endpoint and click on the button to realize request against dhis2 server. The response toString method should be rendered on the screen

How to execute web sample:

1. Open chrome disabling web security:

For OSX:
```
$ open -a Google\ Chrome --args --disable-web-security --user-data-dir
```
--user-data-dir required on Chrome 49+ on OSX

For Linux run:
```
$ google-chrome --disable-web-security
```

For Windows go into the command prompt and go into the folder where Chrome.exe is and type:
```
chrome.exe --disable-web-security
```

2. Compile code to js and create bundle.js

execute:
```
./gradlew buildJsToWeb
```
3. Open web sample

### :floppy_disk: Requires DB migration?


### :art: UI changes?
